File: control-assurance-sp/src/types/IAPActionUpdate.ts
import { IEntity } from "./Entity";


export interface IIAPActionUpdate  extends IEntity {

    UpdateType?: string;
    UpdateDetails?: string;
    RevisedDate?: Date;
    EvFileName?: string;
    EvIsLink?:boolean;
    IAPActionId?: number;
    IAPStatusTypeId?: number;
    UpdatedById?: number;
    UpdateDate?: Date;
    

}

export class IAPActionUpdate implements IIAPActionUpdate{ 
    public ID: number = 0;
    public Title: string = null;    

    public UpdateType?: string = null;
    public UpdateDetails?: string = null;
    public RevisedDate?: Date = null;
    public EvFileName?: string = null;
    public EvIsLink?:boolean = false;
    public IAPActionId?: number = null;
    public IAPStatusTypeId?: number = null;
    public UpdatedById?: number = null;
    public UpdateDate?: Date = null;

    constructor(iapActionId: number, updateType: string) {
        this.IAPActionId = iapActionId;
        this.UpdateType = updateType;
    }


}

File: ControlAssuranceAPI/Models/IAPActionUpdate.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class IAPActionUpdate
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public string? UpdateType { get; set; }

    public string? UpdateDetails { get; set; }

    public DateTime? RevisedDate { get; set; }

    public string? EvFileName { get; set; }

    public bool? EvIsLink { get; set; }

    public int? IAPActionId { get; set; }

    public int? IAPStatusTypeId { get; set; }

    public int? UpdatedById { get; set; }

    public DateTime? UpdateDate { get; set; }

    public virtual IAPAction? IAPAction { get; set; }

    public virtual IAPStatusType? IAPStatusType { get; set; }

    public virtual User? UpdatedBy { get; set; }
}

public static class IAPActionUpdateTypes
{
    public readonly static string ActionUpdate = "Action Update";
    public readonly static string RevisedDate = "Revised Date";
    public readonly static string MiscComment = "Misc Comment";
}

public class IAPActionUpdateView_Result
{
    public int ID { get; set; }
    public string? Title { get; set; }
    public string? UpdateType { get; set; }
    public string? UpdateBy { get; set; }
    public string? UpdateDate { get; set; }
    public string? UpdateDetails { get; set; }
    public string? Status { get; set; }
    public string? RevisedDate { get; set; }
    public string? Evidence { get; set; }
    public bool EvIsLink { get; set; }
}